### PR TITLE
Add Miner property to IActionContext

### LINKED ISSUE #173: Add MinerAddress to ActionContext
Currently, since there is no miner information in the `ActionContext`, there is no way to know who is a miner.

### PR DESCRIPTION
**Pull request by earlbread**
- Added `Miner` property to `IActionContext`.
- Renamed `Block<T>.RewardBeneficiary` to `Block<T>.Miner`
Resolves #173

### CODE REVIEW

**Comment by kijun** [comment]
Perhaps we should call miner block producer instead? I don't have a strong opinion yet, but thinking we might rely on PoA or PoS to produce blocks in the future.

**Comment by dahlia** [comment]
> Perhaps we should call miner block producer instead? I don't have a strong opinion yet, but thinking we might rely on PoA or PoS to produce blocks in the future.

This term (“mine” or “miner”) is currently used everywhere, e.g., `BlockChain<T>.MineBlock()`, `Block<T>.Mine()`.  Even if we decide to reword them we'd better do that in a separated patch.

**Comment by kijun** [comment]
Makes sense!

File: Libplanet/Blocks/Block.cs [hunk 1]
```diff
@@ -173,7 +173,8 @@ public byte[] ToBencodex(bool hash, bool transactionData)
                 );
             foreach (Transaction<T> tx in Transactions)
             {
-                delta = tx.EvaluateActions(Hash, Index, delta);
+                delta = tx.EvaluateActions(
+                    Hash, Index, delta, RewardBeneficiary.Value);
```
Comment (longfin) [inline Libplanet/Blocks/Block.cs]: It seems that `rewardBeneficiary` is no longer a proper name anymore, how about fixing it this time?

**Comment by dahlia** [comment]
FYI:

```
$ rg -i '\bminer?\b|\bminer?[A-Z]'
Libplanet.Tests/TestUtils.cs
78:        internal static Block<T> MineGenesis<T>()
96:        internal static Block<T> MineNext<T>(
116:                return Block<T>.Mine(

Libplanet/Action/IActionContext.cs
45:        /// then reproduce the same result, while neither a single block miner

Libplanet.Tests/Blocks/BlockFixture.cs
16:            Genesis = TestUtils.MineGenesis<PolymorphicAction<BaseAction>>();
17:            Next = TestUtils.MineNext(
25:            HasTx = TestUtils.MineNext(

Libplanet/Blockchain/BlockChain.cs
232:        public Block<T> MineBlock(
254:                Block<T> block = Block<T>.Mine(
272:        public Block<T> MineBlock(Address rewardBeneficiary) =>
273:            MineBlock(rewardBeneficiary, DateTimeOffset.UtcNow);
406:        // ALL blocks (referenced by MineBlock(), etc.) will be changed.

Libplanet/Blocks/Block.cs
104:        public static Block<T> Mine(

Libplanet/Tx/Transaction.cs
274:        /// a <see cref="Libplanet.Blocks.Block{T}"/> is mined.

Libplanet.Tests/Store/FileStoreFixture.cs
77:            Block1 = TestUtils.MineGenesis<DumbAction>();
78:            Block2 = TestUtils.MineNext(Block1);
79:            Block3 = TestUtils.MineNext(Block2);

Libplanet/Blockchain/Policies/IBlockPolicy.cs
11:    /// for a <see cref="Block{T}"/> to be mined.
39:        /// for a new <see cref="Block{T}"/> to be mined
43:        /// followed by a new <see cref="Block{T}"/> to be mined.</param>
45:        /// for a new <see cref="Block{T}"/> to be mined.</returns>

Libplanet.Tests/Blockchain/BlockChainTest.cs
39:            Block<DumbAction> block = _blockChain.MineBlock(_fx.Address1);
43:            Block<DumbAction> anotherBlock = _blockChain.MineBlock(_fx.Address2);
53:            Block<DumbAction> block = _blockChain.MineBlock(_fx.Address1);
56:            Block<DumbAction> anotherBlock = _blockChain.MineBlock(_fx.Address2);
64:            _blockChain.MineBlock(_fx.Address1);
121:            var lastBlock = chain.MineBlock(_fx.Address1);
151:            chain.MineBlock(_fx.Address1);
171:            var block1 = _blockChain.MineBlock(_fx.Address1);
172:            var block2 = _blockChain.MineBlock(_fx.Address1);
173:            var block3 = _blockChain.MineBlock(_fx.Address1);
198:            var block1 = _blockChain.MineBlock(_fx.Address1);
199:            var block2 = _blockChain.MineBlock(_fx.Address1);
200:            var block3 = _blockChain.MineBlock(_fx.Address1);
212:                .Select(_ => _blockChain.MineBlock(_fx.Address1))
249:            chain.MineBlock(_fx.Address1);

Libplanet.Tests/Blocks/BlockTest.cs
47:            Block<PolymorphicAction<BaseAction>> next = MineNext(_fx.Genesis);
180:            Block<DumbAction> genesis = MineGenesis<DumbAction>();
181:            Block<DumbAction> blockIdx1 = MineNext(
215:            Block<DumbAction> blockIdx2 = MineNext(
288:            Block<DumbAction> invalidBlock = MineNext(
289:                MineGenesis<DumbAction>(),
321:            Block<DumbAction> invalidBlock = MineNext(
322:                MineGenesis<DumbAction>(),
359:            Block<PolymorphicAction<BaseAction>> invalidBlock = MineNext(
375:            var block = Block<DumbAction>.Mine(
419:                transactions: MineGenesis<DumbAction>().Transactions
449:                transactions: MineGenesis<DumbAction>().Transactions

Docs/articles/design.md
42:[daemon] runnable on miners' machines (e.g., Linux).
78:when the block is mined.  It is also known as [consensus]
81:It takes 5 to 20 seconds to mine a block.  It means that if we group four

Libplanet.Tests/Net/SwarmTest.cs
165:                        var block = chain.MineBlock(_fx1.Address1);
167:                            $"Block mined. " +
178:            var minerCanceller = new CancellationTokenSource();
179:            Task miningA = CreateMiner(a, chainA, 5000, minerCanceller.Token);
180:            Task miningB = CreateMiner(b, chainB, 8000, minerCanceller.Token);
191:                minerCanceller.Cancel();
416:            Block<DumbAction> genesis = chainA.MineBlock(_fx1.Address1);
418:            Block<DumbAction> block1 = chainA.MineBlock(_fx1.Address1);
419:            Block<DumbAction> block2 = chainA.MineBlock(_fx1.Address1);
483:            chainB.MineBlock(_fx1.Address1);
528:            chainA.MineBlock(_fx1.Address1);
572:            Block<DumbAction> genesis = chainA.MineBlock(_fx1.Address1);
578:                chainA.MineBlock(_fx1.Address1);
584:                chainB.MineBlock(_fx2.Address1);

Libplanet.Tests/Blockchain/Policies/BlockPolicyTest.cs
59:            Block<DumbAction>[] blocks = MineBlocks(
101:                    MineBlocks(
116:                    MineBlocks(
131:                    MineBlocks(
150:                    MineBlocks(
158:                    MineBlocks(
166:                    MineBlocks(
174:                    MineBlocks(
185:                    MineBlocks(
205:                    MineBlocks(
215:                    MineBlocks(
223:        private IEnumerable<Block<DumbAction>> MineBlocks(
228:            _output.WriteLine($"MineBlocks({blockArgs.Length}):");
229:            var miner = default(Address);
254:                Block<DumbAction> block = Block<DumbAction>.Mine(
257:                    miner,
```

🤪

File: Libplanet/Blocks/Block.cs (hunk 1 above)
Comment (dahlia) [reply Libplanet/Blocks/Block.cs]: I'm for this proposal.  How about `Block<T>.Miner`?

Comment (earlbread) [reply Libplanet/Blocks/Block.cs]: I agree. `Block<T>.Miner` seems good.

**Comment by dahlia** [comment]
@kijun Could you create a new issue suggesting to change the term “mine”/“miner”?  We should discuss about that for a long run.

**Comment by earlbread** [comment]
Added a commit to rename `Block<T>.RewardBeneficiary` to `Block<T>.Miner`. @dahlia @longfin

File: CHANGES.md [hunk 2]
```diff
@@ -178,6 +178,8 @@ To be released.
  -  `BencodexFormatter` became able to serialize `BigInteger`.  [[#159]]
  -  Made `Swarm` possible to configure its network `appProtocolVersion` and,
     to ignore peers if their version is different.  [[#167]], [[#170]]
+ -  Added `IActionContext.Miner` property and renamed
+    `Block<T>.RewardBeneficiary` to `Block<T>.Miner`[[#173]], [[#174]]
```
Comment (dahlia) [inline CHANGES.md]: Could you separate this into two bullets?  To me two changes seem independent each other.

Comment (dahlia) [reply CHANGES.md]: Also this statement lacks a period in the end.

Comment (earlbread) [reply CHANGES.md]: I fixed change log and amended commit.

**Pull request merged**